exchange/rpc.go

```diff
@@ -390,8 +390,8 @@ func (p PutDeviceRequest) ShortString() string {
 // Please patch one field at a time.
 type PatchDeviceRequest struct {
 	UserInput          *[]policy.UserInput `json:"userInput,omitempty"`
-	Pattern            string              `json:"pattern,omitempty"`
-	Arch               string              `json:"arch,omitempty"`
+	Pattern            *string             `json:"pattern,omitempty"`
+	Arch               *string             `json:"arch,omitempty"`
```

Comment:
Dont you have to update the String() function to correctly dump the string values, since you've changed them to pointers?

Reply:
These changes (changing string to string pointer) actually were for another PR which sets the pattern to an empty string at the shutdown process. The old way uses PUT, I switch it to PATCH, so we need to change it to *string to distinguish between string not set and set for an empty string. Here I simply copy the code so that I do not have to modify the String() and ShortString() functions after the merge.

Reply:
done